manual_tests/internal/tui/remote_storage_url_form.go

```diff
@@ -0,0 +1,153 @@
+package tui
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/bsv-blockchain/go-wallet-toolbox-manual-tests/internal/fixtures"
+	"github.com/charmbracelet/bubbles/textinput"
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/charmbracelet/lipgloss"
+)
+
+const Placeholder = "http://localhost:8100"
+
+type RemoteStorageURLForm struct {
+	manager   ManagerInterface
+	textInput textinput.Model
+	focus     *FocusManager
+	err       error
+}
+
+func NewRemoteStorageURLForm(manager ManagerInterface) *RemoteStorageURLForm {
+	ti := textinput.New()
+	ti.Placeholder = Placeholder
+	ti.CharLimit = 256
+	ti.Width = 50
+
+	form := &RemoteStorageURLForm{
+		manager:   manager,
+		textInput: ti,
+		focus:     NewFocusManager(),
+		err:       nil,
+	}
+
+	// Set up focus items: URL input, Continue, Back
+	form.focus.SetItems([]FocusItem{
+		{Type: ElementInput, Index: 0, Label: "Server URL"},
+		{Type: ElementButton, Index: ButtonContinue, Label: "Connect"},
+		{Type: ElementButton, Index: ButtonBack, Label: fixtures.ButtonBack},
+	})
+
+	form.updateInputFocus()
+	return form
+}
+
+func (m *RemoteStorageURLForm) updateInputFocus() {
+	m.textInput.Blur()
+
+	current := m.focus.CurrentItem()
+	if current.Type == ElementInput {
+		m.textInput.Focus()
+	}
+}
+
+func (m *RemoteStorageURLForm) Init() tea.Cmd {
+	return textinput.Blink
+}
+
+func (m *RemoteStorageURLForm) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
+	switch msg := msg.(type) {
+	case tea.KeyMsg:
+		switch msg.Type {
+		case tea.KeyCtrlC, tea.KeyEsc:
+			return m, tea.Quit
+		case tea.KeyEnter:
+			return m.handleEnter()
+
+		case tea.KeyShiftTab, tea.KeyCtrlP, tea.KeyUp:
+			m.focus.Previous()
+			m.updateInputFocus()
+		case tea.KeyTab, tea.KeyCtrlN, tea.KeyDown:
+			m.focus.Next()
+			m.updateInputFocus()
+		}
+	}
+
+	current := m.focus.CurrentItem()
+	var cmd tea.Cmd
+	if current.Type == ElementInput {
+		m.textInput, cmd = m.textInput.Update(msg)
+	} else {
+		cmd = nil
+	}
+
+	return m, cmd
```

Comment:
redundant return nil directly instead of assigning and then returning ....
```suggestion
	current := m.focus.CurrentItem()
	if current.Type == ElementInput {
		var cmd tea.Cmd
		m.textInput, cmd = m.textInput.Update(msg)
		return m, cmd
	}

	return m, nil
```

Reply:
changed